### PR TITLE
Index license type on dependency registry

### PR DESCRIPTION
**Pull request by yoshiwarab**
## Description of the change

This PR is base off of https://github.com/ArtBlocks/artblocks-subgraph/pull/254 and includes updates to index license type on the Dependency entity type. Additionally it includes changes for compatibility with the latest updates to the DependencyRegistryV0 contract (see https://github.com/ArtBlocks/artblocks-contracts/pull/1215).

>reminder: Any subgraph deployments should be documented as [Releases](https://github.com/ArtBlocks/artblocks-subgraph/releases) in this repository.

>reminder: Any changes to subgraph schema should be accompanied by corresponding changes to the Art Blocks documentation site, available at https://docs.artblocks.io/ (specifically, see the [Subgraph Entities](https://docs.artblocks.io/creator-docs/art-blocks-api/entities/) and [Subgraph Querying and API Overview](https://docs.artblocks.io/creator-docs/art-blocks-api/queries/) sections).

### CODE REVIEW

File: src/dependency-registry.ts [hunk 1]
```diff
@@ -367,7 +415,7 @@ export function handleSupportedCoreContractRemoved(
   event: SupportedCoreContractRemoved
 ): void {
   const coreContract = Contract.load(
```
Comment (ryley-o) [inline src/dependency-registry.ts:417]: unrelated to this PR - probably worth a check that `coreContract.dependencyRegistry == event.address` before setting it to null? (I know we only expect one dependency registry, but who knows...)

File: src/dependency-registry.ts [hunk 2]
```diff
@@ -379,42 +427,42 @@ export function handleSupportedCoreContractRemoved(
...
+    event.params.projectId
   );
 
   const project = Project.load(projectId);
   if (!project) {
     return;
   }
 
-  project.scriptTypeAndVersion = event.params._dependencyType.toString();
+  project.scriptTypeAndVersion = event.params.dependencyNameAndVersion.toString();
```
Comment (ryley-o) [inline src/dependency-registry.ts:443]: Also unrelated to this specific PR, but I don't think we should allow both core contract to set this field, and also the core registry. The state in the subgraph then depends on ordering, which might be pretty confusing? WDYT about a follow-on PR to do something like add a nullable `scriptTypeAndVersionOverride` field to project?

**Review by ryley-o — APPROVED** [review]
These changes make sense to me!
Added a couple notes on things that I think are worth touching on, but I think follow-on work since this PR is meant for naming updates.

File: src/dependency-registry.ts (hunk 2 above)
Comment (yoshiwarab) [reply src/dependency-registry.ts:443]: Ah, yes definitely agree!

File: src/dependency-registry.ts (hunk 1 above)
Comment (yoshiwarab) [reply src/dependency-registry.ts:417]: Agreed! https://app.asana.com/0/1201568815538912/1206099136189468/f

**Pull request merged**